src/models/NetworkAppController.js

```diff
@@ -152,7 +159,7 @@ export default class NetworkAppController {
     const networkStdlib = this.networkPackage.stdlib;
     const hasNetworkStdlib = !_.isEmpty(networkStdlib);
     const hasCustomDeploy = hasNetworkStdlib && networkStdlib.customDeploy;
-    const customDeployMatches = hasCustomDeploy && networkStdlib.name === this.package.stdlib.name;
+    const customDeployMatches = hasCustomDeploy && networkStdlib.name === this.package.stdlib.name && networkStdlib.version === this.package.stdlib.version;
```

Comment:
we can extract these behaviors to some functions like `hasSameNameAndVersion` or so